app/questionnaire/navigation.py

```diff
@@ -34,147 +37,128 @@ def build_navigation(self, current_group_id, current_group_instance):
         if navigation_block is None or navigation_block.get('visible', True) is False:
             return None
 
-        summary_or_confirmation_blocks = self.schema.get_summary_and_confirmation_blocks()
-
         navigation = []
 
         for section in self.schema.sections:
             non_skipped_groups = self._get_non_skipped_groups(section)
             if not non_skipped_groups:
                 continue
 
-            first_group_id = non_skipped_groups[0]
-            first_block_id = self.schema.get_first_block_id_for_group(first_group_id)
+            target_location = self._get_location_for_section(section, non_skipped_groups)
+
+            is_summary_or_confirmation_section = (
+                QuestionnaireSchema.is_summary_section(section) or
+                QuestionnaireSchema.is_confirmation_section(section)
+            )
 
-            if first_block_id in summary_or_confirmation_blocks \
-                    and self._can_reach_summary_confirmation(first_group_id, first_block_id) is False:
+            if is_summary_or_confirmation_section and \
+                    not self._can_reach_summary_confirmation(section):
                 continue
 
-            repeating_rule = self.schema.get_repeat_rule(
-                self.schema.get_group(first_group_id))
+            # if the first group in a section is a repeating group then repeat the section
+            # navigation link for each repeat instead of rendering the section title
+            repeating_rule = self.schema.get_repeat_rule(non_skipped_groups[0])
```

Comment:
I think we should also change this to sections that contain one 1 group that repeats.